webhooks/pod_webhook.go

```diff
@@ -45,6 +45,12 @@ type PodMutator struct {
 
 // Handle injects the flagd sidecar (if the prerequisites are all met)
 func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	defer func() {
```

Comment:
I wanted to create a test to capture this behaviour. Unfortunately without this block the test would pass regardless of whether the nil check was there or not. The code would panic but because the panic wasn't being caught it'd fail silently (in the eyes of the test).
This safeguard might be overly cautious, perhaps including a test for this case is overkill and just including the nil fix would suffice.
I couldn't come to a decision on this, some other opinions would be appreciated @AlexsJones @toddbaert @beeme1mr @james-milligan

Reply:
The panic can't be caught at a higher level (e.g. in the test suite) because `recover()` can't catch panics in goroutines.

Reply:
I think it's overkill personally, but I commend your efforts to improve things.

Reply:
I think it probably is too, my main concern is that something like this can slip through the cracks and not raise any alarms with test failures.